### PR TITLE
Submitting the galaxy jobs as a predefined system user: using 'real_system_username'

### PR DESCRIPTION
**Pull request by ashvark**
Up to now, there are two possible ways of specifying the **real_system_username**  in galaxy.ini for submitting the jobs as the actual system user.  This PR facilitates the third possible value which can be a name of the real system user who will run all the jobs being submitted. This user should not the user running the galaxy system. 

I think this option is necessary, because jobs submitted as the same user running the galaxy system has full permissions over all the dataset files.  Whereas, jobs submitted as real system user has an advantage that it requires only read access to all dataset files. Most of the galaxy servers (or atlease our system) do not have real system users matching galaxy user_email or username. This PR would of helpfull for these people if they feel insecure in running the jobs as galaxy's system user. 

@natefoo @erasche We discussed this in gcc2017. Let me know your comments on this PR

### CODE REVIEW

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -204,8 +204,12 @@ def system_user_pwent(self, real_system_username):
             except KeyError:
                 pass
         else:
```
Comment (bernt-matthias) [inline lib/galaxy/model/__init__.py:206]: I think you should test for `real_system_username is not None`, i.e. the else branch should be an elif branch. Or is this function never called if real_system_username is not configured? The default is None, isn't it?

Comment (ashvark) [reply lib/galaxy/model/__init__.py:206]: @bernt-matthias  I think the default value for 'real_system_username' is 'user_email'. This function 'system_user_pwent' is called only when 'external_runjob_script' is set in galaxy.ini. Do I still need to check for None ?

Comment (bernt-matthias) [reply lib/galaxy/model/__init__.py:206]: You are right.

**Comment by jmchilton** [comment]
Looks good to me - thanks for the patch!

I think it would be better to setup Pulsar if this is a serious concern - you could get proper separation of the users and you could get much more structured access to the data. But I'm incredibly biased and the approach presented here seems totally viable.

**Pull request merged**